modules/claims_api/app/controllers/claims_api/v2/application_controller.rb

```diff
@@ -160,6 +168,21 @@ def build_target_veteran(veteran_id:, loa:) # rubocop:disable Metrics/MethodLeng
         target_veteran[:va_profile] = ClaimsApi::Veteran.build_profile(mpi_profile.birth_date)
         target_veteran
       end
+
+      def validate_id_with_icn(bgs_claim, lighthouse_claim, request_icn)
+        claim_prtcpnt_id = if bgs_claim&.dig(:benefit_claim_details_dto).present?
+                             bgs_claim&.dig(:benefit_claim_details_dto, :ptcpnt_vet_id)
+                           end
+        veteran_icn = if lighthouse_claim.present? && lighthouse_claim['veteran_icn'].present?
+                        lighthouse_claim['veteran_icn']
+                      end
+
+        if claim_prtcpnt_id != target_veteran.participant_id && veteran_icn != request_icn
+          raise ::Common::Exceptions::ResourceNotFound.new(
+            detail: 'Invalid claim ID for the veteran identified.'
+          )
+        end
+      end
```

Comment:
interesting that we moved this in this PR? was this needed or just in preparation for another change?